tests/rptest/tests/leadership_transfer_test.py

```diff
@@ -139,18 +139,41 @@ def all_partitions_present(nodes: int):
 
             return True
 
+        def count_leaders_per_node(topic_name: str):
+            leaders_per_node = collections.defaultdict(int)
+            tps = self.redpanda.partitions(topic_name)
+            for p in tps:
+                if p.leader:
+                    leaders_per_node[p.leader] += 1
+
+            return leaders_per_node
+
+        def distribution_error():
+            nodes = [
+                self.redpanda.node_id(n)
+                for n in self.redpanda.started_nodes()
+            ]
+            error = 0.0
+            for t in self.topics:
+                replicas = t.partition_count * 3
+                leaders_per_node = count_leaders_per_node(topic_name=t.name)
+                replicas_per_node = replicas / len(nodes)
```

Comment:
a bit confused about replicas_per_node.. I think for topic_aware_balancer constraint the error function for a given node `n` is  (leaders[n] - optimal_leaders[n])^2 where opt_leaders = total_partitions/total_nodes_with_replicas .. I think this loop is doing something different, no? (especially replicas_per_node part).

Reply:
you are right, i was skewed by partition balancing, changing this